### PR TITLE
mrp lint ci

### PR DESCRIPTION
**Pull request by lshamis**
# Description

Add MRP lint to CI

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

If this PR introduces a change or fixes a bug, please add before and after screenshots (if this is causes a UX change) or before and after examples(this could be plain text, videos etc ) to demonstrate the change.

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

# Testing

Please describe the tests that you ran to verify your changes and how we can reproduce.

# Checklist:

- [ ] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

**Review by exhaustin — APPROVED** [review]
CircleCI test looks good.

I made the following changes:
- Add pre-commit hook that auto-formats changes
- Update README with instructions
- Format repo using black

**Comment by exhaustin** [comment]
One concern: flake8 is complaining about certain examples assigning values to variables that weren't being used afterwards. I feel like "meaningless" assignments like this within examples still convey information about the API to the user, thus I changed CircleCI to only lint the src code instead of going in and fixing the examples. Thoughts?

File: mrp/src/mrp/cmd/up.py [hunk 1]
```diff
@@ -79,13 +79,13 @@ def callback(system_state):
                 ns.sat = True
                 ns.cv.notify()
 
-    watcher = life_cycle.system_state_watcher(callback)
+    life_cycle.system_state_watcher(callback)
```
Comment (lshamis) [inline mrp/src/mrp/cmd/up.py:82]: The `watcher` variable is required. It manages the lifetime of the thread that executes the callback.

File: mrp/src/mrp/cmd/wait.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def callback(sys_state):
             ns.sat = True
             ns.cv.notify()
 
-    watcher = life_cycle.system_state_watcher(callback)
+    life_cycle.system_state_watcher(callback)
```
Comment (lshamis) [inline mrp/src/mrp/cmd/wait.py:32]: See above comment.

File: mrp/src/mrp/cmd/up.py (hunk 1 above)
Comment (exhaustin) [reply mrp/src/mrp/cmd/up.py:82]: Thanks for catching it. However `flake8` views `watcher` as an unused variable and complains about it in the test. What do you think is the best way to handle this?

Comment (exhaustin) [reply mrp/src/mrp/cmd/up.py:82]: Maybe I can add an ignore list to ignore these type of rules?

Comment (alephzero) [reply mrp/src/mrp/cmd/up.py]: Can we just annotate that one line:
```py
watcher = life_cycle.system_state_watcher(callback)  # noqa: F841
```

Comment (exhaustin) [reply mrp/src/mrp/cmd/up.py]: My concern is that all future uses of watcher will then need to have that annotation, not sure if this is scalable.

**Pull request merged**